ext/reflection/php_reflection.c

```diff
@@ -2953,27 +2953,49 @@ ZEND_METHOD(reflection_method, __construct)
 	char *name_str, *tmp;
 	size_t name_len, tmp_len;
 	zval ztmp;
+	const char *space;
 
-	if (zend_parse_parameters_ex(ZEND_PARSE_PARAMS_QUIET, ZEND_NUM_ARGS(), "zs", &classname, &name_str, &name_len) == FAILURE) {
-		if (zend_parse_parameters_throw(ZEND_NUM_ARGS(), "s", &name_str, &name_len) == FAILURE) {
-			return;
-		}
+	if (zend_parse_parameters_throw(ZEND_NUM_ARGS(), "z|s!", &classname, &name_str, &name_len) == FAILURE) {
+		return;
+	}
 
-		if ((tmp = strstr(name_str, "::")) == NULL) {
-			zend_throw_exception_ex(reflection_exception_ptr, 0,
-				"Invalid method name %s", name_str);
-			return;
-		}
-		classname = &ztmp;
-		tmp_len = tmp - name_str;
-		ZVAL_STRINGL(classname, name_str, tmp_len);
-		name_len = name_len - (tmp_len + 2);
-		name_str = tmp + 2;
-		orig_obj = NULL;
-	} else if (Z_TYPE_P(classname) == IS_OBJECT) {
-		orig_obj = classname;
-	} else {
-		orig_obj = NULL;
+	switch (Z_TYPE_P(classname)) {
+		case IS_STRING:
+			orig_obj = NULL;
+
+			if (ZEND_NUM_ARGS() == 1) {	
+				name_str = Z_STRVAL_P(classname);
+				name_len = Z_STRLEN_P(classname);
+
+				if ((tmp = strstr(name_str, "::")) == NULL) {
+					zend_throw_exception_ex(reflection_exception_ptr, 0, "Invalid method name %s", name_str);
+					return;
+				}
+
+				classname = &ztmp;
+				tmp_len = tmp - name_str;
+				ZVAL_STRINGL(classname, name_str, tmp_len);
+				name_len = name_len - (tmp_len + 2);
+				name_str = tmp + 2;
+			}
+
+			break;
+
+		case IS_OBJECT:
+			if (ZEND_NUM_ARGS() == 1 || !name_str) {
```

Comment:
```suggestion
			if (!name_str) {
```
`name_str` will be `NULL` if 1 argument is given

Reply:
I had to explicitly put `ZEND_NUM_ARGS() == 1 ||` because when no parameter was provided, `name_str` was filled up with garbage. Despite i had been looking why for a very long time, I still ignore the reason...

Reply:
You have to initialize `name_str` and `name_len`, since optional parameters are only initialized during ZPP if they are provided. So that's the reason :)

Reply:
Oh, thank you! My C is still a little rusty after so much time...

Reply:
I know what you're talking about :) I needed quite a lot of time to have even a vague idea about what I see in php-src :)

Reply:
> 
> 
> Oh, thank you! My C is still a little rusty after so much time...

PHP-src is more it's own version of C than *actual* C at times due to the bazillion macros, so yeah don't worry lol